src/Integration.Vsix/Suppression/DelegateInjector.cs

```diff
@@ -42,8 +42,22 @@ namespace SonarLint.VisualStudio.Integration.Vsix.Suppression
     /// </summary>
     internal sealed class DelegateInjector : IDisposable
     {
-        public DelegateInjector(Func<Diagnostic, bool> function)
+        private readonly Func<Diagnostic, bool> suppressionFunction;
+        private readonly IServiceProvider serviceProvider;
+
+        public DelegateInjector(Func<Diagnostic, bool> suppressionFunction, IServiceProvider serviceProvider)
         {
+            if (suppressionFunction == null)
+            {
+                throw new ArgumentNullException(nameof(suppressionFunction));
```

Comment:
I might have missed something but I haven't seen any test of these 2 `ArgumentNullException` cases.

Reply:
Added